src/System Application/Test/Cryptography Management/src/RSATest.Codeunit.al

```diff
@@ -222,22 +222,34 @@ codeunit 132617 "RSA Test"
         EncryptingOutStream: OutStream;
         EncryptedInStream: InStream;
         EncryptedOutStream: OutStream;
+        DecryptedInStream: InStream;
         DecryptedOutStream: OutStream;
+        PlainText: Text;
+        DecryptedText: Text;
+        DecryptionFailed: Boolean;
     begin
         // [SCENARIO] Decrypt text encrypted with use of PKCS#1 padding, using OAEP padding.
+        // [SCENARIO] Due to random padding, decryption may occasionally not throw but returns garbage data.
         Initialize();
 
         // [GIVEN] With RSA pair of keys, plain text and encryption stream
         EncryptingTempBlob.CreateOutStream(EncryptingOutStream);
-        SaveRandomTextToOutStream(EncryptingOutStream);
+        PlainText := SaveRandomTextToOutStream(EncryptingOutStream);
         EncryptingTempBlob.CreateInStream(EncryptingInStream);
         EncryptedTempBlob.CreateOutStream(EncryptedOutStream);
         RSA.Encrypt(PrivateKeyXmlStringSecret, EncryptingInStream, false, EncryptedOutStream);
         EncryptedTempBlob.CreateInStream(EncryptedInStream);
 
         // [WHEN] Decrypt encrypted text stream using OAEP Padding
         DecryptingTempBlob.CreateOutStream(DecryptedOutStream);
-        asserterror RSA.Decrypt(PrivateKeyXmlStringSecret, EncryptedInStream, true, DecryptedOutStream);
+        DecryptionFailed := not TryDecrypt(RSA, PrivateKeyXmlStringSecret, EncryptedInStream, true, DecryptedOutStream);
+
+        // [THEN] Either decryption fails with an exception, or the decrypted text is garbage (not equal to plaintext)
+        if not DecryptionFailed then begin
+            DecryptingTempBlob.CreateInStream(DecryptedInStream);
+            DecryptedText := Base64Convert.FromBase64(Base64Convert.ToBase64(DecryptedInStream));
+            LibraryAssert.AreNotEqual(PlainText, DecryptedText, 'Decryption with wrong padding should fail or return garbage data.');
```

Comment:
your message here should only be about garbage data no? if encryption fails you will never get to this assert